x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/action/TransportMonitoringMigrateAlertsAction.java

```diff
@@ -0,0 +1,235 @@
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.monitoring.action.MonitoringMigrateAlertsAction;
+import org.elasticsearch.xpack.core.monitoring.action.MonitoringMigrateAlertsRequest;
+import org.elasticsearch.xpack.core.monitoring.action.MonitoringMigrateAlertsResponse;
+import org.elasticsearch.xpack.core.monitoring.action.MonitoringMigrateAlertsResponse.ExporterMigrationResult;
+import org.elasticsearch.xpack.monitoring.Monitoring;
+import org.elasticsearch.xpack.monitoring.exporter.Exporter;
+import org.elasticsearch.xpack.monitoring.exporter.Exporter.ExporterResourceStatus;
+import org.elasticsearch.xpack.monitoring.exporter.Exporters;
+import org.elasticsearch.xpack.monitoring.exporter.MonitoringMigrationCoordinator;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+public class TransportMonitoringMigrateAlertsAction extends TransportMasterNodeAction<MonitoringMigrateAlertsRequest,
+    MonitoringMigrateAlertsResponse> {
+
+    private static final Logger logger = LogManager.getLogger(TransportMonitoringMigrateAlertsAction.class);
+
+    private final Client client;
+    private final MonitoringMigrationCoordinator migrationCoordinator;
+    private final Exporters exporters;
+
+    @Inject
+    public TransportMonitoringMigrateAlertsAction(Client client, Exporters exporters, MonitoringMigrationCoordinator migrationCoordinator,
+                                                  TransportService transportService, ClusterService clusterService, ThreadPool threadPool,
+                                                  ActionFilters actionFilters, IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(MonitoringMigrateAlertsAction.NAME, transportService, clusterService, threadPool, actionFilters,
+            MonitoringMigrateAlertsRequest::new, indexNameExpressionResolver, MonitoringMigrateAlertsResponse::new,
+            ThreadPool.Names.MANAGEMENT);
+        this.client = client;
+        this.migrationCoordinator = migrationCoordinator;
+        this.exporters = exporters;
+    }
+
+    @Override
+    protected void masterOperation(Task task, MonitoringMigrateAlertsRequest request, ClusterState state,
+                                   ActionListener<MonitoringMigrateAlertsResponse> listener) throws Exception {
+        // First, enable the migration coordinator block
+        if (migrationCoordinator.tryBlockInstallationTasks() == false) {
+            throw new EsRejectedExecutionException("Could not migrate cluster alerts. Timed out waiting to block resource operations.");
```

Comment:
nit: there is no more time out.  

Also, can we change the language from "block" to "currently running"  ?